### ISSUE TITLE
Vanilla Snapshots Encountering Fatal Errors "No Such File or Directory"

### ISSUE THREAD

**Issue by AverageCakeSlice** [itzg/docker-minecraft-server#549]
I'm trying to spin up a container with the latest snapshot jar, but for some reason it refuses to launch for me. I had a snapshot up and running previously, but ever since the pre-release of 1.16, none of the snapshots (including older ones like 20w19a) are working.  

---
  
Here are the logs that are output when the server crashes:

> [init] Running as uid=1000 gid=1000 with /data as 'drwxrwxrwx 1 1000 1000 4096 Jun 5 11:08 /data'
> 
> [init] Resolved version given SNAPSHOT into 1.16-pre2
> 
> [init] Resolving type given VANILLA
> 
> [init] server.properties already created, skipping
> 
> [init] Setting/adding ops
> 
> [init] log4j2.xml already created, skipping
> 
> [init] Checking for JSON files.
> 
> [init] Setting initial memory to 1G and max to 4G
> 
> [init] Starting the Minecraft server...
> 
> [11:08:59] [main/ERROR]: Unknown biome, defaulting to plains
> 
> [11:09:01] [main/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', name='PROD'
> 
> Exception in thread "main" [11:09:01] [main/INFO]: [STDERR]: java.lang.Error: java.io.IOException: No such file or directory
> 
> [11:09:01] [main/INFO]: [STDERR]: at sun.nio.ch.FileKey.create(FileKey.java:46)
> 
> [11:09:01] [main/INFO]: [STDERR]: at sun.nio.ch.SharedFileLockTable.<init>(FileLockTable.java:120)
> 
> [11:09:01] [main/INFO]: [STDERR]: at sun.nio.ch.FileLockTable.newSharedFileLockTable(FileLockTable.java:47)
> 
> [11:09:01] [main/INFO]: [STDERR]: at sun.nio.ch.FileChannelImpl.fileLockTable(FileChannelImpl.java:1039)
> 
> [11:09:01] [main/INFO]: [STDERR]: at sun.nio.ch.FileChannelImpl.tryLock(FileChannelImpl.java:1106)
> 
> [11:09:01] [main/INFO]: [STDERR]: at java.nio.channels.FileChannel.tryLock(FileChannel.java:1155)
> 
> [11:09:01] [main/INFO]: [STDERR]: at ado.a(SourceFile:42)
> 
> [11:09:01] [main/INFO]: [STDERR]: at daa$a.<init>(SourceFile:280)
> 
> [11:09:01] [main/INFO]: [STDERR]: at daa.c(SourceFile:268)
> 
> [11:09:01] [main/INFO]: [STDERR]: at net.minecraft.server.Main.main(SourceFile:118)
> 
> [11:09:01] [main/INFO]: [STDERR]: Caused by: java.io.IOException: No such file or directory
> 
> [11:09:01] [main/INFO]: [STDERR]: at sun.nio.ch.FileKey.init(Native Method)
> 
> [11:09:01] [main/INFO]: [STDERR]: at sun.nio.ch.FileKey.create(FileKey.java:44)
> 
> [11:09:01] [main/INFO]: [STDERR]: ... 9 more
> 
> 2020-06-05T11:09:01.719-0400 INFO mc-server-runner Done

**Comment by AverageCakeSlice** [comment]
Also, here are the contents of my docker-compose file: 
```
minecraft-server:
    ports:
      - "25567:25565"
 
    environment:
      EULA: "TRUE"
      TZ: "America/Indiana/Indianapolis"
      TYPE: "VANILLA"
      VERSION: "SNAPSHOT"
      DIFFICULTY: "normal"
      SPAWN_PROTECTION: 0
      OPS: "AverageCakeSlice"
      INIT_MEMORY: "1G"
      MAX_MEMORY: "4G"

    volumes:
      - "C:/Users/Bread/_containers/gameboise/_data:/data"

    image: itzg/minecraft-server
  
    container_name: gameboise
  
    tty: true
    stdin_open: true
    restart: always
```

**Comment by itzg** [comment]
Unfortunately the exception it threw was pretty vague since it didn't seem to state what file wasn't found. FWIW, I was able to start a fresh snapshot container:

https://gist.github.com/itzg/bcd2315a1d658c0af14b5dc04a76146b#file-gistfile1-txt

I'll try later with an existing container / data directory that I have running on a previous snapshot.

**Comment by AverageCakeSlice** [comment]
Huh, I was able to execute the command that you just did, and was able to do it with the detach flag. It must be something in my compose file or on my local machine. Thanks.

**Comment by seannik** [comment]
Same problem occurred with me, same log data.  Seems to be related to attaching data directory to local system (-v <windows path> and using WSL2 as the docker engine.
My machine with docker using WSL2 ran fine when removing only the "-v <windows path>" argument and another one running Docker still using Hyper-V engine runs fine with the "-v <windows path>".  Hope this helps.

**Comment by itzg** [comment]
@seannik sorry for not providing this info on the issue, but this is caused by the upstream issue https://bugs.mojang.com/browse/MC-177336

**Comment by seannik** [comment]
no worries

**Comment by itzg** [comment]
FYI, the upstream issue is now fixed in `1.16-pre3`
